libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -123,17 +145,18 @@ pub(super) fn validate_index_names(ctx: &mut Context<'_>) {
         .iter_models()
         .flat_map(|(model_id, _)| ctx.db.types.models[&model_id].indexes.iter())
     {
-        let index_name: &str = match index.name {
-            Some(index_name) => index_name,
+        let index_name = match &index.db_name {
+            Some(index_name) => index_name.to_string(),
             None => continue,
         };
 
-        if index_names.insert(index_name) {
+        if index_names.insert(index_name.clone()) {
```

Comment:
that's a bad side effect of the `Cow`'s — but we already said we'd clean up later so let's keep it for now